src/App/Fossa/ListTargets.hs

```diff
@@ -97,14 +100,50 @@ runAll ::
   , Has (Reader AllFilters) sig m
   , Has Telemetry sig m
   ) =>
+  ListTargetOutputFormat ->
   Path Abs Dir ->
   m ()
-runAll basedir = traverse_ single discoverFuncs
+runAll outputFmt basedir = traverse_ single discoverFuncs
   where
-    single (DiscoverFunc f) = withDiscoveredProjects f basedir (printSingle basedir)
+    single (DiscoverFunc f) = withDiscoveredProjects f basedir formatter
+    formatter proj = case outputFmt of
+      Legacy -> renderLegacyFmt basedir proj
+      Text -> renderTextFmt basedir proj
+      NdJSON -> renderNdJson basedir proj
 
-printSingle :: (ToJSON a, Has Logger sig m) => Path Abs Dir -> DiscoveredProject a -> m ()
-printSingle basedir project = do
+-- | Render targets in ndjson format
+-- Reference: https://github.com/ndjson/ndjson-spec
+renderNdJson :: (Has Logger sig m, Has (Lift IO) sig m) => Path Abs Dir -> DiscoveredProject a -> m ()
+renderNdJson basedir project = do
+  let maybeRel = makeRelative basedir (projectPath project)
+  let render txt = logStdout . decodeUtf8 $ (encode txt) <> "\n"
+  case maybeRel of
+    Nothing -> pure ()
+    Just rel ->
+      case projectBuildTargets project of
+        ProjectWithoutTargets ->
+          render $ object ["type" .= projectType project, "path" .= toFilePath rel]
+        FoundTargets targets -> for_ (Set.toList $ toSet targets) $ \target -> do
```

Comment:
Sets are Foldable so I think that `Set.toList` is unnecessary, `for_` should work just fine with it.